addons/account/tests/test_early_payment_discount.py

```diff
@@ -681,11 +677,11 @@ def test_mixed_epd_with_tax_no_duplication(self):
             ],
             'invoice_payment_term_id': self.early_pay_10_percents_10_days.id,
         })
-        self.assertEqual(len(inv.line_ids), 6) # 1 prod, 1 tax, 2 epd, 1 epd tax discount, 1 payment terms
+        self.assertEqual(len(inv.line_ids), 5)  # 1 prod, 1 tax, 1 epd, 1 epd tax discount, 1 payment terms
```

Comment:
@qdp-odoo The epd line were generating some tax lines with the `epd` type. Previously, having:
product 100
epd -10
... were generating 2 tax lines:
tax: 21
tax: - 2.1 (linked to the epd)

This behavior was problematic because the tax line itself doesn't know if it was created for an epd line or not. This was leading to useless recomputation and complexify the generic method to create the tax lines (`_prepare_tax_lines`). For the story, I unified back the computation of tax lines in the invoices and in others object like the bank reconciliation widget. By doing that, I had to fix the inconsistencies and the extra epd tax line was one of them. I checked with CHKL about that and he was ok to merge the 2 tax lines in one.